packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -89,13 +89,21 @@ Future<io.WebSocket> _defaultOpenChannel(String url, {
   io.WebSocket socket;
 
   Future<void> handleError(dynamic e) async {
-    globals.printTrace('Exception attempting to connect to Observatory: $e');
-    globals.printTrace('This was attempt #$attempts. Will retry in $delay.');
-
+    void Function(String) printVisibleTrace = globals.printTrace;
     if (attempts == 10) {
-      globals.printStatus('This is taking longer than expected...');
+      globals.printStatus('Connecting to the Observatory is taking longer than expected...');
+    } else if (attempts == 20) {
+      globals.printStatus('Still attempting to connect to the Observatory...');
+      globals.printStatus(
+        'Consider running with --host-vmservice-port to use a specific port '
+        'that is known to be available.');
+    } else if (attempts % 50 == 0) {
```

Comment:
why not `attempts == 50`?

Reply:
My preference is to have a message that prints periodically (but slowly) if it's still attempting to do work so that it does not appear to be a hung process.  In practice I expect that most people will give up waiting fairly soon, so I doubt it would matter much either way.